### ISSUE TITLE
feat: support list groups in enhanced `Select`

### ISSUE THREAD

**Issue by nicholaschiang** [jamesmfriedman/rmwc#614]
Right now, there's no way to add `ListDivider`s to an enhanced `Select` component (without building the component yourself using a composition of a `readOnly` controlled `TextField` and `Menu`).

It'd be nice to be able to do something like this:
```jsx
<Select enhanced options=[
  { label: 'Option #1', value: 1 }, 
  { type: 'divider' }, 
  { label: 'Option #2', value: 2, type: 'option' },
] />
```
Where the `type` property would be optional (and would default to `option`).

See [this issue](https://github.com/tutorbookapp/covid-tutoring/issues/14) for my current application (i.e. how this feature would be used).

**Comment by jamesmfriedman** [comment]
They're already supported actually. Straight from the docs

```
<Select
  label="Formatted"
  enhanced
  options={[
    {
      label: 'Dinner',
      options: [
        {
          label: 'Pizza',
          value: '2'
        }
      ]
    },
    {
      label: 'Dessert',
      options: [
        {
          label: 'Cookies',
          value: '1'
        },

        {
          label: 'Icecream',
          value: '3'
        }
      ]
    }
  ]}
/>
```

If you don't want the section labels, just pass empty strings.

**Comment by nicholaschiang** [comment]
Yeah @jamesmfriedman, but it looks like it still inserts a `<div class="mdc-list-group__subheader" />` which I don't want:

![Screenshot 2020-04-21 at 1 52 01 PM](https://user-images.githubusercontent.com/20798889/79912822-7efdf000-83d7-11ea-832b-2f9ccacc8b61.png)

Those subheaders make the divisions between the lists uneven (i.e. it looks like there's a lot more padding before the list than after it to the end user).

**Comment by nicholaschiang** [comment]
Could you make it omit those subheaders when we pass a `null` or empty string `label`?
